src/driver/DriverUtils.ts

```diff
@@ -127,14 +127,16 @@ export class DriverUtils {
         buildOptions: { shorten?: boolean; joiner?: string } | undefined,
         ...alias: string[]
     ): string {
-        const newAlias =
-            alias.length === 1
-                ? alias[0]
-                : alias.join(
-                      buildOptions && buildOptions.joiner
-                          ? buildOptions.joiner
-                          : "_",
-                  )
+        const joiner =
+            buildOptions && buildOptions.joiner ? buildOptions.joiner : "_"
+
+        let newAlias = alias.length === 1 ? alias[0] : alias.join(joiner)
+
+        const hasNestedRelations = newAlias.includes(".")
```

Comment:
Is there some kind of constant that defines this as the relations joiner in typeorm? I've skimmed through the repo and didn't find one. Maybe you have more insight. 😊

Reply:
That's some nice insight. I do think it should be defined somewhere in the repo, otherwise, where is it taking it from?. I'll see if I can find it, maybe we can work the solution through there 👀

Reply:
I think there are many places in the code which just assumes that there are no dots in the name of tables.
Maybe worth another issue. There is already #3531. Maybe there somewhere hides a function that escapes names or something. However, not sure if this is out of scope for this PR.

Reply:
Just made an update to the PR. By using relationName instead of relationPath, we do not need to parse for any "." to get the relations. So we won't encounter issues with the table names. (At least not in this section, it seems the full support for them is still missing in the repo😅 )